packages/react-components/react-popover/e2e/Popover.e2e.tsx

```diff
@@ -378,5 +378,28 @@ describe('Popover', () => {
         cy.contains('Two').should('have.focus');
       });
     });
+
+    describe('trap focus behaviour', () => {
+      it('should focus on PopoverSurface when its tabIndex is a number', () => {
```

Comment:
Please also include steps to tab out and back into the `PopoverSurface` from the `document` to confirm that the `PopoverSurface` does not stay within tab order, which was the recommendation @smhigley  if I remember correctly